integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/database/device/DlmsDeviceSteps.java

```diff
@@ -786,7 +787,42 @@ private ProtocolInfo getProtocolInfo(final Map<String, String> inputSettings) {
         inputSettings.getOrDefault(
             PlatformSmartmeteringKeys.PROTOCOL_VERSION,
             PlatformSmartmeteringDefaults.PROTOCOL_VERSION);
-    return this.protocolInfoRepository.findByProtocolAndProtocolVersion(protocol, protocolVersion);
+
+    final String port =
+        inputSettings.getOrDefault(
+            PlatformSmartmeteringKeys.PORT, PlatformSmartmeteringDefaults.PORT.toString());
+    final ProtocolInfo protocolInfo =
+        this.protocolInfoRepository.findByProtocolAndProtocolVersion(protocol, protocolVersion);
+    if (protocolInfo == null) {
+      throw new IllegalArgumentException(
+          String.format(
+              "No protocol info found with combination of protocol %s and version %s",
+              protocol, protocolVersion));
+    }
+    this.checkPortAndProtocolMatch(port, protocolInfo);
```

Comment:
Zou je het ook nog zo kunnen doen dat als er geen port gezet is, hij niet de default pakt, maar die uit de port mapping?

Reply:
Changing the input setting in the glue steps does not seem okay since we do not know where these settings will be used